streams/src/test/java/org/apache/kafka/streams/kstream/RepartitionTopicNamingTest.java

```diff
@@ -94,24 +94,49 @@ public void shouldFailWithSameRepartitionTopicName() {
         }
     }
 
-    // each KGroupedStream will result in repartition, can't reuse
-    // KGroupedStreams when specifying repartition topic names and Optimization is turned off
-    // need to have separate groupByKey calls when naming repartition topics
-    // see test shouldHandleUniqueGroupedInstances below for an example
     @Test
-    public void shouldFailWithSameRepartitionTopicNameUsingSameKGroupedStream() {
-        try {
-            final StreamsBuilder builder = new StreamsBuilder();
-            final KGroupedStream<String, String> kGroupedStream = builder.<String, String>stream("topic").selectKey((k, v) -> k).groupByKey(Grouped.as("grouping"));
-            kGroupedStream.windowedBy(TimeWindows.of(Duration.ofMillis(10L))).count();
-            kGroupedStream.windowedBy(TimeWindows.of(Duration.ofMillis(30L))).count();
-            builder.build();
-            fail("Should not build re-using repartition topic name");
-        } catch (final TopologyException te) {
-            // ok
-        }
+    public void shouldNotFailWithSameRepartitionTopicNameUsingSameKGroupedStream() {
+        final StreamsBuilder builder = new StreamsBuilder();
+        final KGroupedStream<String, String> kGroupedStream = builder.<String, String>stream("topic").selectKey((k, v) -> k).groupByKey(Grouped.as("grouping"));
+        kGroupedStream.windowedBy(TimeWindows.of(Duration.ofMillis(10L))).count().toStream().to("output-one");
+        kGroupedStream.windowedBy(TimeWindows.of(Duration.ofMillis(30L))).count().toStream().to("output-two");
+        final String topologyString = builder.build().describe().toString();
+        assertThat(1, is(getCountOfRepartitionTopicsFound(topologyString, repartitionTopicPattern)));
+        assertTrue(topologyString.contains("grouping" + "-repartition"));
```

Comment:
nit: why string concatenation? simplify to `"grouping-repartition"`

Reply:
ack